views.py

```diff
@@ -195,3 +191,196 @@ def view_custom_page(request, page_name):
+            if 'autofill_ror' in request.POST:
+                ror = supporter.get_ror()
+                if ror:
+                    if ror == supporter.ror:
+                        messages.add_message(
+                            request,
+                            messages.SUCCESS,
+                            f'Existing ROR confirmed: { ror }.'
+                        )
+                    else:
+                        messages.add_message(
+                            request,
+                            messages.SUCCESS,
+                            f'Autofilled new ROR: { ror }.'
+                        )
+                    supporter.ror = ror
+                    supporter.save()
+                    supporter_form = forms.EditSupporterForm(
+                        instance=supporter,
+                    )
+                else:
+                    messages.add_message(
+                        request,
+                        messages.WARNING,
+                        f'ROR could not be autofilled.'
+                    )
+            elif 'remove_contact' in request.POST:
+                contact = supporter_models.SupporterContact.objects.get(
+                    pk=request.POST.get('remove_contact')
+                )
+                message = f'Contact removed: { contact }.'
+                contact.delete()
+                messages.add_message(request, messages.SUCCESS, message)
+            elif 'add_contact' in request.POST:
+                account = Account.objects.get(pk=request.POST.get('add_contact'))
+                contact, _created = supporter_models.SupporterContact.objects.get_or_create(
+                    supporter=supporter,
+                    account=account,
+                )
+                message = f'Contact added: { contact }.'
+                messages.add_message(request, messages.SUCCESS, message)
+            elif 'add_band' in request.POST:
+                add_band = True
+            elif 'save_continue' in request.POST:
+                user_search_form = forms.AccountAdminSearchForm()
+            elif 'save_return' in request.POST and next_url:
+                return redirect(next_url)
+            elif 'search_user' in request.POST or request.POST['q']:
+                user_search_form = forms.AccountAdminSearchForm(request.POST)
+                results, _dups = search_model_admin(request, Account)
+                user_search_results = results.exclude(
+                    supportercontact__supporter=supporter,
+                )[:10]
+
+            if not supporter_id:
+                return redirect(
+                    reverse(
+                        'edit_supporter_band',
+                        kwargs={'supporter_id': supporter.pk}
+                    )
+                )
+
+        supporter_form = forms.EditSupporterForm(instance=supporter)
+        band_form = forms.EditBandForm(instance=band)
+
+    template = 'consortial_billing/edit_supporter_band.html'
```

Comment:
I went back and forth on this when writing the original view. I acknowledge the branching is unwieldy. The main goal was to make a single page where you can control all of the information related to a supporter. Because we do not have a standard way to refresh part of the page in Janeway, I needed all the handling to be done through one view where all the relevant forms are checked and saved with each user action, because I did not want any data entered to be lost on refresh, regardless of whether the user clicks on the "correct" button. Once we have an async solution, I think we can split the view up into four views, one for each of supporter form, ror form, contact form, and band form.